packages/edit-site/src/components/sidebar-navigation-screen-global-styles/index.js

```diff
@@ -155,26 +85,49 @@ function SidebarNavigationScreenGlobalStylesFooter( { onClickRevisions } ) {
 export default function SidebarNavigationScreenGlobalStyles() {
 	const { openGeneralSidebar, setIsListViewOpened } =
 		useDispatch( editSiteStore );
+	const { createInfoNotice } = useDispatch( noticesStore );
+	const { set: setPreference } = useDispatch( preferencesStore );
 	const isMobileViewport = useViewportMatch( 'medium', '<' );
 	const { setCanvasMode, setEditorCanvasContainerView } = unlock(
 		useDispatch( editSiteStore )
 	);
+	const browseStylesLink = useLink( {
+		path: '/wp_global_styles/browseStyles',
+	} );
+	const { isStyleBookOpened, hasGlobalStyleVariations, isDistractionFree } =
+		useSelect(
+			( select ) => ( {
+				isStyleBookOpened:
+					'style-book' ===
+					unlock(
+						select( editSiteStore )
+					).getEditorCanvasContainerView(),
+				hasGlobalStyleVariations:
+					!! select(
+						coreStore
+					).__experimentalGetCurrentThemeGlobalStylesVariations()
+						?.length,
+				isDistractionFree: select( preferencesStore ).get(
+					editSiteStore.name,
+					'distractionFree'
+				),
+			} ),
+			[]
+		);
 
-	const isStyleBookOpened = useSelect(
-		( select ) =>
-			'style-book' ===
-			unlock( select( editSiteStore ) ).getEditorCanvasContainerView(),
-		[]
-	);
-
-	const openGlobalStyles = useCallback(
-		async () =>
-			Promise.all( [
-				setCanvasMode( 'edit' ),
-				openGeneralSidebar( 'edit-site/global-styles' ),
-			] ),
-		[ setCanvasMode, openGeneralSidebar ]
-	);
+	const openGlobalStyles = useCallback( async () => {
+		// Disable distraction free mode.
+		if ( isDistractionFree ) {
+			setPreference( editSiteStore.name, 'distractionFree', false );
```

Comment:
Why is this better than toggle feature?

Reply:
Because if you don't have enabled DFM and `toggle`, you end up in DFM, and that is not what we want.
Additionally `toggleFeature` is a deprecated action.

Reply:
I think I tought that `false` was toggling it to false. Welp I better check for other cases of misuse!